lib/hydra/works/services/generic_file/full_text_extraction_service.rb

```diff
@@ -30,19 +37,31 @@ def extract
     # @return [String] the result of calling the extract service
     def fetch
       req = Net::HTTP.new(uri.host, uri.port)
-      resp = req.post(uri.to_s, original_file.content, request_headers)
+      resp = req.post(uri.to_s, file_content, request_headers)
       raise Hydra::Works::FullTextExtractionError.new, "Solr Extract service was unsuccessful. '#{uri}' returned code #{resp.code} for #{id}\n#{resp.body}" unless resp.code == '200'
-      original_file.content.rewind if original_file.content.respond_to?(:rewind)
+      file_content.rewind if file_content.respond_to?(:rewind)
 
       resp.body
     end
 
+    def file_content
+      @content ||= if file_path
+                     File.open(file_path).read
+                   else
+                     original_file.content
+                   end
+    end
+
     # @return [Hash] the request headers to send to the Solr extract service
     def request_headers
-      { Faraday::Request::UrlEncoded::CONTENT_TYPE => "#{original_file.mime_type};charset=utf-8",
```

Comment:
Can `original_file.mime_type` return `nil` now? If so, do things break? The new code is explicitly OK with returning `nil` and I'm wondering if our test coverage of this condition needs to be expanded. If this does not change current behavior, I'm :+1: on this PR.